client/my-sites/site-settings/jetpack-credentials/index.jsx

```diff
@@ -14,58 +13,49 @@ import CompactCard from 'components/card/compact';
 import CredentialsSetupFlow from './credentials-setup-flow';
 import CredentialsConfigured from './credentials-configured';
 import Gridicon from 'gridicons';
-import QueryRewindStatus from 'components/data/query-rewind-status';
-import QueryJetpackCredentials from 'components/data/query-jetpack-credentials';
+import QueryRewindState from 'components/data/query-rewind-state';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { hasMainCredentials, isRewindActive } from 'state/selectors';
+import { getRewindState } from 'state/selectors';
 
 class JetpackCredentials extends Component {
-	static propTypes = {
-		hasMainCredentials: PropTypes.bool,
-		isRewindActive: PropTypes.bool,
-		siteId: PropTypes.number.isRequired,
-	};
-
 	render() {
-		const {
-			hasMainCredentials, // eslint-disable-line no-shadow
-			isRewindActive, // eslint-disable-line no-shadow
-			translate,
-			siteId,
-		} = this.props;
+		const { credentials, rewindState, siteId, translate } = this.props;
+		const hasAuthorized = rewindState === 'provisioning' || rewindState === 'active';
+		const hasCredentials = credentials && credentials.length > 0;
```

Comment:
We're really only concerned about having main role credentials here. Maybe try something like

`const hasCredentials = 'undefined' !== typeof find( credentials, { 'role': 'main' } )`

Reply:
It is theoretically possible that a site will/can have alternate credentials but not main creds.

Reply:
Updated in 768e39b25f

To me this is mounting evidence that _for now_ what might be most helpful is to only expose main credentials to the clients. We don't currently support multiple credentials and I don't see that happening soon. Maybe it's no big deal, but we might want to start gauging if it could simplify multiple spots if we just made `credentials` a property that's either the main credentials or nothing.